File: Projeto06_c++/Projeto06_c++/src/UsuarioAutenticavel.cpp
#include "../include/UsuarioAutenticavel.h"

UsuarioAutenticavel::UsuarioAutenticavel() : Usuario(), senha("") {}

UsuarioAutenticavel::UsuarioAutenticavel(const std::string& nome, const std::string& email, const std::string& tipo, const std::string& senha)
    : Usuario(nome, email, tipo), senha(senha) {}

bool UsuarioAutenticavel::verificarSenha(const std::string& senha) const {
    return this->senha == senha;
}

File: Projeto_6/Projeto-6_KW/src/usuarioAutenticavel.py
from abc import ABC, abstractmethod

class UsuarioAutenticavel(ABC):
    def __init__(self, nome: str, senha: str):
        self._nome = nome
        self.__senha = senha  # senha encapsulada

    def _verificar_senha(self, senha: str) -> bool:
        return self.__senha == senha

    @abstractmethod
    def autenticar(self, senha: str) -> bool:
        pass

    @property
    def nome(self):
        return self._nome